Build failure: python313Packages.weasyprint

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

`nix-build -I nixpkgs=. -A python313Packages.weasyprint` on branch nixos-unstable.

Note that the same issue is raised with Python 3.13 and Python 3.14.

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/318173913

### Relevant log output

```console
this derivation will be built:
  /nix/store/1zfm396yb39h8rp3yybhm2h28y6llpvk-python3.14-weasyprint-66.0.drv
building '/nix/store/1zfm396yb39h8rp3yybhm2h28y6llpvk-python3.14-weasyprint-66.0.drv'...
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-build-hook
Using pypaBuildPhase
Sourcing python-runtime-deps-check-hook
Using pythonRuntimeDepsCheckHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
Using versionCheckHook
Running phase: unpackPhase
unpacking source archive /nix/store/k9phz9knmpc4vda6xpmhvq7dg5khbbc1-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file "source/weasyprint/urls.py"
Running phase: patchPhase
applying patch /nix/store/a3qdbk1xxj4l66y9lha7i99mk1rb5p41-library-paths.patch
patching file weasyprint/text/ffi.py
Hunk #1 succeeded at 472 (offset 28 lines).
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...
* Building wheel...
Successfully built weasyprint-66.0-py3-none-any.whl
Finished creating a wheel...
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for weasyprint-66.0-py3-none-any.whl
Finished executing pythonRuntimeDepsCheck
Running phase: installPhase
Executing pypaInstallPhase
Successfully installed weasyprint-66.0-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: fixupPhase
shrinking RPATHs of ELF executables and libraries in /nix/store/x870ay4yr4km98iiai23rkdr0jy8p3dw-python3.14-weasyprint-66.0
checking for references to /build/ in /nix/store/x870ay4yr4km98iiai23rkdr0jy8p3dw-python3.14-weasyprint-66.0...
patching script interpreter paths in /nix/store/x870ay4yr4km98iiai23rkdr0jy8p3dw-python3.14-weasyprint-66.0
stripping (with command strip and flags -S -p) in  /nix/store/x870ay4yr4km98iiai23rkdr0jy8p3dw-python3.14-weasyprint-66.0/lib /nix/store/x870ay4yr4km98iiai23rkdr0jy8p3dw-python3.14-weasyprint-66.0/bin
shrinking RPATHs of ELF executables and libraries in /nix/store/9a5dj7j8cx5yflzr5ixcm08nl2lpd01d-python3.14-weasyprint-66.0-dist
checking for references to /build/ in /nix/store/9a5dj7j8cx5yflzr5ixcm08nl2lpd01d-python3.14-weasyprint-66.0-dist...
patching script interpreter paths in /nix/store/9a5dj7j8cx5yflzr5ixcm08nl2lpd01d-python3.14-weasyprint-66.0-dist
Rewriting #!/nix/store/pmb8clh66mryha2ijzz3dg969drrkgj7-python3-3.14.2/bin/python3.14 to #!/nix/store/pmb8clh66mryha2ijzz3dg969drrkgj7-python3-3.14.2
wrapping `/nix/store/x870ay4yr4km98iiai23rkdr0jy8p3dw-python3.14-weasyprint-66.0/bin/weasyprint'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
Executing versionCheckPhase
Successfully managed to find version 66.0 in the output of the command /nix/store/x870ay4yr4km98iiai23rkdr0jy8p3dw-python3.14-weasyprint-66.0/bin/weasyprint --version

Fontconfig error: No writable cache directories
        /.cache/fontconfig
        /var/cache/fontconfig

Fontconfig error: No writable cache directories
        /.cache/fontconfig
        /var/cache/fontconfig

Fontconfig error: No writable cache directories
        /.cache/fontconfig
        /var/cache/fontconfig

Fontconfig error: No writable cache directories
        /.cache/fontconfig
        /var/cache/fontconfig
WeasyPrint version 66.0
Finished versionCheckPhase
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
Running phase: pythonRemoveBinBytecodePhase
Running phase: pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: weasyprint
Running phase: pytestCheckPhase
Executing pytestCheckPhase
pytest flags: -m pytest -k not\ \(test_font_stretch\)\ and\ not\ \(test_linear_gradients_12\)\ and\ not\ \(test_linear_gradients_5\)\ and\ not\ \(test_tab_size\)\ and\ not\ \(test_tabulation_character\)\ and\ not\ \(test_empty_inline_auto_margins\)\ and\ not\ \(test_images_transparent_text\)\ and\ not\ \(test_layout_table_auto_44\)\ and\ not\ \(test_layout_table_auto_45\)\ and\ not\ \(test_margin_boxes_element\)\ and\ not\ \(test_running_elements\)\ and\ not\ \(test_vertical_align_4\)\ and\ not\ \(test_visibility_1\)\ and\ not\ \(test_visibility_3\)\ and\ not\ \(test_visibility_4\)\ and\ not\ \(test_woff_simple\)
============================= test session starts ==============================
platform linux -- Python 3.14.2, pytest-8.4.2, pluggy-1.6.0
rootdir: /build/source
configfile: pyproject.toml
plugins: cov-nixpkgs-stub-1.0.0
collected 3298 items / 27 deselected / 3271 selected                           

tests/css/test_common.py ...............                                 [  0%]
tests/css/test_counters.py ........x......................               [  1%]
tests/css/test_descriptors.py ................                           [  1%]
tests/css/test_errors.py ............                                    [  2%]
tests/css/test_expanders.py ............................................ [  3%]
........................................................................ [  5%]
........................................................................ [  8%]
........................................................................ [ 10%]
.....................................................                    [ 11%]
tests/css/test_fonts.py .....................                            [ 12%]
tests/css/test_nesting.py ........                                       [ 12%]
tests/css/test_pages.py ........................                         [ 13%]
tests/css/test_target.py .......                                         [ 13%]
tests/css/test_ua.py ...                                                 [ 13%]
tests/css/test_validation.py ........................................... [ 15%]
........................................................................ [ 17%]
........................................................................ [ 19%]
........................................................................ [ 21%]
........................................................................ [ 23%]
........................................................................ [ 26%]
........................................................................ [ 28%]
........................................................................ [ 30%]
........................................................................ [ 32%]
........................................................................ [ 34%]
........................................................................ [ 37%]
....................................................                     [ 38%]
tests/css/test_variables.py ............................................ [ 40%]
........................................................................ [ 42%]
........................................................................ [ 44%]
..............................................                           [ 45%]
tests/draw/svg/test_bounding_box.py .........xx..                        [ 46%]
tests/draw/svg/test_clip.py ..x                                          [ 46%]
tests/draw/svg/test_defs.py ..                                           [ 46%]
tests/draw/svg/test_gradients.py ..............................          [ 47%]
tests/draw/svg/test_images.py ........x....                              [ 47%]
tests/draw/svg/test_markers.py ........                                  [ 47%]
tests/draw/svg/test_opacity.py ..xx...                                   [ 48%]
tests/draw/svg/test_paths.py ......................                      [ 48%]
tests/draw/svg/test_patterns.py .......                                  [ 49%]
tests/draw/svg/test_shapes.py ...................                        [ 49%]
tests/draw/svg/test_text.py ........................                     [ 50%]
tests/draw/svg/test_transform.py ..................                      [ 50%]
tests/draw/svg/test_units.py ....                                        [ 51%]
tests/draw/svg/test_visibility.py ........                               [ 51%]
tests/draw/test_absolute.py ............x...............                 [ 52%]
tests/draw/test_background.py .......................................... [ 53%]
...............                                                          [ 53%]
tests/draw/test_before_after.py ...                                      [ 53%]
tests/draw/test_box.py .................................                 [ 54%]
tests/draw/test_column.py ....                                           [ 55%]
tests/draw/test_current_color.py .....x..                                [ 55%]
tests/draw/test_float.py ............xx.............x.x..                [ 56%]
tests/draw/test_footnote.py ...........                                  [ 56%]
tests/draw/test_footnote_column.py .........x                            [ 56%]
tests/draw/test_gradient.py .................                            [ 57%]
tests/draw/test_image.py ............................................... [ 58%]
.....                                                                    [ 59%]
tests/draw/test_leader.py ....xx.x..........                             [ 59%]
tests/draw/test_list.py ...                                              [ 59%]
tests/draw/test_opacity.py ......                                        [ 59%]
tests/draw/test_overflow.py .........                                    [ 60%]
tests/draw/test_page.py ............                                     [ 60%]
tests/draw/test_table.py .....................x........x....             [ 61%]
tests/draw/test_text.py .......x.....x..............xxx.........X...     [ 62%]
tests/draw/test_transform.py FFFF..........                              [ 63%]
tests/draw/test_visibility.py .                                          [ 63%]
tests/draw/test_whitespace.py .........                                  [ 63%]
tests/layout/test_block.py ............................................. [ 65%]
................x...........                                             [ 65%]
tests/layout/test_column.py ............................................ [ 67%]
...                                                                      [ 67%]
tests/layout/test_flex.py .............................................. [ 68%]
........................................................................ [ 70%]
....................................................                     [ 72%]
tests/layout/test_float.py ..........................x..                 [ 73%]
tests/layout/test_footnotes.py ...x......................                [ 74%]
tests/layout/test_grid.py ..............................                 [ 75%]
tests/layout/test_image.py ............................................. [ 76%]
......................                                                   [ 77%]
tests/layout/test_inline.py .x...........x.............................. [ 78%]
......                                                                   [ 78%]
tests/layout/test_inline_block.py ..                                     [ 78%]
tests/layout/test_list.py .......x..                                     [ 79%]
tests/layout/test_page.py ..................................x........... [ 80%]
..........................................................               [ 82%]
tests/layout/test_position.py ................                           [ 82%]
tests/layout/test_preferred.py ......................................... [ 84%]
..................................................                       [ 85%]
tests/layout/test_table.py ............................................. [ 86%]
..........x............................................................. [ 89%]
..............                                                           [ 89%]
tests/test_acid2.py .                                                    [ 89%]
tests/test_api.py ...................................................... [ 91%]
.....................                                                    [ 91%]
tests/test_boxes.py .................................................... [ 93%]
.......                                                                  [ 93%]
tests/test_fonts.py ......                                               [ 93%]
tests/test_pdf.py ...............................................        [ 95%]
tests/test_presentational_hints.py .........                             [ 95%]
tests/test_stacking.py ........                                          [ 95%]
tests/test_text.py ..................................................... [ 97%]
........................................................................ [ 99%]
.....x..                                                                 [ 99%]
tests/test_unicode.py .                                                  [ 99%]
tests/test_url.py ...                                                    [100%]

=================================== FAILURES ===================================
_____________________________ test_2d_transform_1 ______________________________

assert_pixels = <function assert_pixels.<locals>.<lambda> at 0x7fffe2736560>

    @assert_no_logs
    def test_2d_transform_1(assert_pixels):
>       assert_pixels('''
            ________
            ________
            __BBBr__
            __BBBB__
            __BBBB__
            __BBBB__
            ________
            ________
        ''', '''
          <style>
            @page { size: 8px; margin: 2px; }
            div { transform: rotate(90deg); font-size: 0 }
          </style>
          <div><img src="pattern.png"></div>''')

tests/draw/test_transform.py:8: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/conftest.py:96: in <lambda>
    return lambda *args, **kwargs: draw.assert_pixels(
tests/draw/__init__.py:58: in assert_pixels
    assert_pixels_equal(name, width, height, pixels, expected_pixels)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

name = '2d_transform_1', width = 8, height = 8
raw = <ImagingCore object at 0x7ffff0ed4670>
expected_raw = [(255, 255, 255), (255, 255, 255), (255, 255, 255), (255, 255, 255), (255, 255, 255), (255, 255, 255), ...]
tolerance = 0

    def assert_pixels_equal(name, width, height, raw, expected_raw, tolerance=0):
        """Take 2 matrices of pixels and assert that they are the same."""
        if raw != expected_raw:  # pragma: no cover
            pixels = zip_longest(raw, expected_raw, fillvalue=(-1, -1, -1))
            for i, (value, expected) in enumerate(pixels):
                if expected is None:
                    continue
                if any(abs(value - expected) > tolerance
                       for value, expected in zip(value, expected)):
                    actual_height = len(raw) // width
                    write_png(name, raw, width, actual_height)
                    expected_raw = [
                        pixel or (255, 255, 255) for pixel in expected_raw]
                    write_png(f'{name}.expected', expected_raw, width, height)
                    x = i % width
                    y = i // width
>                   assert 0, (
                           ^
                        f'Pixel ({x}, {y}) in {name}: '
                        f'expected rgba{expected}, got rgba{value}')
E                   AssertionError: Pixel (2, 1) in 2d_transform_1: expected rgba(255, 255, 255), got rgba(0, 0, 255)

tests/draw/__init__.py:107: AssertionError
------------------------------ Captured log call -------------------------------
INFO     weasyprint.progress:__init__.py:165 Step 1 - Fetching and parsing HTML - HTML string
INFO     weasyprint.progress:__init__.py:292 Step 2 - Fetching and parsing CSS - CSS string
INFO     weasyprint.progress:__init__.py:69 Step 3 - Applying CSS
INFO     weasyprint.progress:document.py:229 Step 4 - Creating formatting structure
INFO     weasyprint.progress:page.py:988 Step 5 - Creating layout - Page 1
INFO     weasyprint.progress:__init__.py:119 Step 6 - Creating PDF
INFO     weasyprint.progress:__init__.py:231 Step 7 - Adding PDF metadata
_____________________________ test_2d_transform_2 ______________________________

assert_pixels = <function assert_pixels.<locals>.<lambda> at 0x7fffe339dfe0>

    @assert_no_logs
    def test_2d_transform_2(assert_pixels):
>       assert_pixels('''
            ____________
            ____________
            _____BBBr___
            _____BBBB___
            _____BBBB___
            _____BBBB___
            ____________
            ____________
            ____________
            ____________
            ____________
            ____________
        ''', '''
          <style>
            @page { size: 12px; margin: 2px; }
            div { transform: translateX(3px) rotate(90deg);
                  font-size: 0; width: 4px }
          </style>
          <div><img src="pattern.png"></div>''')

tests/draw/test_transform.py:27: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/conftest.py:96: in <lambda>
    return lambda *args, **kwargs: draw.assert_pixels(
tests/draw/__init__.py:58: in assert_pixels
    assert_pixels_equal(name, width, height, pixels, expected_pixels)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

name = '2d_transform_2', width = 12, height = 12
raw = <ImagingCore object at 0x7fffe3516a70>
expected_raw = [(255, 255, 255), (255, 255, 255), (255, 255, 255), (255, 255, 255), (255, 255, 255), (255, 255, 255), ...]
tolerance = 0

    def assert_pixels_equal(name, width, height, raw, expected_raw, tolerance=0):
        """Take 2 matrices of pixels and assert that they are the same."""
        if raw != expected_raw:  # pragma: no cover
            pixels = zip_longest(raw, expected_raw, fillvalue=(-1, -1, -1))
            for i, (value, expected) in enumerate(pixels):
                if expected is None:
                    continue
                if any(abs(value - expected) > tolerance
                       for value, expected in zip(value, expected)):
                    actual_height = len(raw) // width
                    write_png(name, raw, width, actual_height)
                    expected_raw = [
                        pixel or (255, 255, 255) for pixel in expected_raw]
                    write_png(f'{name}.expected', expected_raw, width, height)
                    x = i % width
                    y = i // width
>                   assert 0, (
                           ^
                        f'Pixel ({x}, {y}) in {name}: '
                        f'expected rgba{expected}, got rgba{value}')
E                   AssertionError: Pixel (5, 1) in 2d_transform_2: expected rgba(255, 255, 255), got rgba(0, 0, 255)

tests/draw/__init__.py:107: AssertionError
------------------------------ Captured log call -------------------------------
INFO     weasyprint.progress:__init__.py:165 Step 1 - Fetching and parsing HTML - HTML string
INFO     weasyprint.progress:__init__.py:292 Step 2 - Fetching and parsing CSS - CSS string
INFO     weasyprint.progress:__init__.py:69 Step 3 - Applying CSS
INFO     weasyprint.progress:document.py:229 Step 4 - Creating formatting structure
INFO     weasyprint.progress:page.py:988 Step 5 - Creating layout - Page 1
INFO     weasyprint.progress:__init__.py:119 Step 6 - Creating PDF
INFO     weasyprint.progress:__init__.py:231 Step 7 - Adding PDF metadata
_____________________________ test_2d_transform_3 ______________________________

assert_pixels = <function assert_pixels.<locals>.<lambda> at 0x7fffe339d640>

    @assert_no_logs
    def test_2d_transform_3(assert_pixels):
        # A translateX after the rotation is actually a translateY
>       assert_pixels('''
            ____________
            ____________
            ____________
            ____________
            ____________
            __BBBr______
            __BBBB______
            __BBBB______
            __BBBB______
            ____________
            ____________
            ____________
        ''', '''
          <style>
            @page { size: 12px; margin: 2px; }
            div { transform: rotate(90deg) translateX(3px);
                  font-size: 0; width: 4px }
          </style>
          <div><img src="pattern.png"></div>''')

tests/draw/test_transform.py:52: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/conftest.py:96: in <lambda>
    return lambda *args, **kwargs: draw.assert_pixels(
tests/draw/__init__.py:58: in assert_pixels
    assert_pixels_equal(name, width, height, pixels, expected_pixels)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

name = '2d_transform_3', width = 12, height = 12
raw = <ImagingCore object at 0x7fffe35149f0>
expected_raw = [(255, 255, 255), (255, 255, 255), (255, 255, 255), (255, 255, 255), (255, 255, 255), (255, 255, 255), ...]
tolerance = 0

    def assert_pixels_equal(name, width, height, raw, expected_raw, tolerance=0):
        """Take 2 matrices of pixels and assert that they are the same."""
        if raw != expected_raw:  # pragma: no cover
            pixels = zip_longest(raw, expected_raw, fillvalue=(-1, -1, -1))
            for i, (value, expected) in enumerate(pixels):
                if expected is None:
                    continue
                if any(abs(value - expected) > tolerance
                       for value, expected in zip(value, expected)):
                    actual_height = len(raw) // width
                    write_png(name, raw, width, actual_height)
                    expected_raw = [
                        pixel or (255, 255, 255) for pixel in expected_raw]
                    write_png(f'{name}.expected', expected_raw, width, height)
                    x = i % width
                    y = i // width
>                   assert 0, (
                           ^
                        f'Pixel ({x}, {y}) in {name}: '
                        f'expected rgba{expected}, got rgba{value}')
E                   AssertionError: Pixel (2, 4) in 2d_transform_3: expected rgba(255, 255, 255), got rgba(0, 0, 255)

tests/draw/__init__.py:107: AssertionError
------------------------------ Captured log call -------------------------------
INFO     weasyprint.progress:__init__.py:165 Step 1 - Fetching and parsing HTML - HTML string
INFO     weasyprint.progress:__init__.py:292 Step 2 - Fetching and parsing CSS - CSS string
INFO     weasyprint.progress:__init__.py:69 Step 3 - Applying CSS
INFO     weasyprint.progress:document.py:229 Step 4 - Creating formatting structure
INFO     weasyprint.progress:page.py:988 Step 5 - Creating layout - Page 1
INFO     weasyprint.progress:__init__.py:119 Step 6 - Creating PDF
INFO     weasyprint.progress:__init__.py:231 Step 7 - Adding PDF metadata
_____________________________ test_2d_transform_4 ______________________________

assert_pixels = <function assert_pixels.<locals>.<lambda> at 0x7fffe2734670>

    @assert_no_logs
    def test_2d_transform_4(assert_pixels):
>       assert_pixels('''
            ____________
            ____________
            ____________
            ____________
            ____________
            __BBBr______
            __BBBB______
            __BBBB______
            __BBBB______
            ____________
            ____________
            ____________
        ''', '''
          <style>
            @page { size: 12px; margin: 2px; }
            div { transform: rotate(90deg); font-size: 0; width: 4px }
            img { transform: translateX(3px) }
          </style>
          <div><img src="pattern.png"></div>''')

tests/draw/test_transform.py:76: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/conftest.py:96: in <lambda>
    return lambda *args, **kwargs: draw.assert_pixels(
tests/draw/__init__.py:58: in assert_pixels
    assert_pixels_equal(name, width, height, pixels, expected_pixels)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

name = '2d_transform_4', width = 12, height = 12
raw = <ImagingCore object at 0x7fffe3516510>
expected_raw = [(255, 255, 255), (255, 255, 255), (255, 255, 255), (255, 255, 255), (255, 255, 255), (255, 255, 255), ...]
tolerance = 0

    def assert_pixels_equal(name, width, height, raw, expected_raw, tolerance=0):
        """Take 2 matrices of pixels and assert that they are the same."""
        if raw != expected_raw:  # pragma: no cover
            pixels = zip_longest(raw, expected_raw, fillvalue=(-1, -1, -1))
            for i, (value, expected) in enumerate(pixels):
                if expected is None:
                    continue
                if any(abs(value - expected) > tolerance
                       for value, expected in zip(value, expected)):
                    actual_height = len(raw) // width
                    write_png(name, raw, width, actual_height)
                    expected_raw = [
                        pixel or (255, 255, 255) for pixel in expected_raw]
                    write_png(f'{name}.expected', expected_raw, width, height)
                    x = i % width
                    y = i // width
>                   assert 0, (
                           ^
                        f'Pixel ({x}, {y}) in {name}: '
                        f'expected rgba{expected}, got rgba{value}')
E                   AssertionError: Pixel (2, 4) in 2d_transform_4: expected rgba(255, 255, 255), got rgba(0, 0, 255)

tests/draw/__init__.py:107: AssertionError
------------------------------ Captured log call -------------------------------
INFO     weasyprint.progress:__init__.py:165 Step 1 - Fetching and parsing HTML - HTML string
INFO     weasyprint.progress:__init__.py:292 Step 2 - Fetching and parsing CSS - CSS string
INFO     weasyprint.progress:__init__.py:69 Step 3 - Applying CSS
INFO     weasyprint.progress:document.py:229 Step 4 - Creating formatting structure
INFO     weasyprint.progress:page.py:988 Step 5 - Creating layout - Page 1
INFO     weasyprint.progress:__init__.py:119 Step 6 - Creating PDF
INFO     weasyprint.progress:__init__.py:231 Step 7 - Adding PDF metadata
=========================== short test summary info ============================
FAILED tests/draw/test_transform.py::test_2d_transform_1 - AssertionError: Pixel (2, 1) in 2d_transform_1: expected rgba(255, 255, 255...
FAILED tests/draw/test_transform.py::test_2d_transform_2 - AssertionError: Pixel (5, 1) in 2d_transform_2: expected rgba(255, 255, 255...
FAILED tests/draw/test_transform.py::test_2d_transform_3 - AssertionError: Pixel (2, 4) in 2d_transform_3: expected rgba(255, 255, 255...
FAILED tests/draw/test_transform.py::test_2d_transform_4 - AssertionError: Pixel (2, 4) in 2d_transform_4: expected rgba(255, 255, 255...
==== 4 failed, 3233 passed, 27 deselected, 33 xfailed, 1 xpassed in 28.64s =====
error: Cannot build '/nix/store/1zfm396yb39h8rp3yybhm2h28y6llpvk-python3.14-weasyprint-66.0.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/9a5dj7j8cx5yflzr5ixcm08nl2lpd01d-python3.14-weasyprint-66.0-dist
         /nix/store/x870ay4yr4km98iiai23rkdr0jy8p3dw-python3.14-weasyprint-66.0
       Last 25 log lines:
       >                         pixel or (255, 255, 255) for pixel in expected_raw]
       >                     write_png(f'{name}.expected', expected_raw, width, height)
       >                     x = i % width
       >                     y = i // width
       > >                   assert 0, (
       >                            ^
       >                         f'Pixel ({x}, {y}) in {name}: '
       >                         f'expected rgba{expected}, got rgba{value}')
       > E                   AssertionError: Pixel (2, 4) in 2d_transform_4: expected rgba(255, 255, 255), got rgba(0, 0, 255)
       >
       > tests/draw/__init__.py:107: AssertionError
       > ------------------------------ Captured log call -------------------------------
       > INFO     weasyprint.progress:__init__.py:165 Step 1 - Fetching and parsing HTML - HTML string
       > INFO     weasyprint.progress:__init__.py:292 Step 2 - Fetching and parsing CSS - CSS string
       > INFO     weasyprint.progress:__init__.py:69 Step 3 - Applying CSS
       > INFO     weasyprint.progress:document.py:229 Step 4 - Creating formatting structure
       > INFO     weasyprint.progress:page.py:988 Step 5 - Creating layout - Page 1
       > INFO     weasyprint.progress:__init__.py:119 Step 6 - Creating PDF
       > INFO     weasyprint.progress:__init__.py:231 Step 7 - Adding PDF metadata
       > =========================== short test summary info ============================
       > FAILED tests/draw/test_transform.py::test_2d_transform_1 - AssertionError: Pixel (2, 1) in 2d_transform_1: expected rgba(255, 255, 255...
       > FAILED tests/draw/test_transform.py::test_2d_transform_2 - AssertionError: Pixel (5, 1) in 2d_transform_2: expected rgba(255, 255, 255...
       > FAILED tests/draw/test_transform.py::test_2d_transform_3 - AssertionError: Pixel (2, 4) in 2d_transform_3: expected rgba(255, 255, 255...
       > FAILED tests/draw/test_transform.py::test_2d_transform_4 - AssertionError: Pixel (2, 4) in 2d_transform_4: expected rgba(255, 255, 255...
       > ==== 4 failed, 3233 passed, 27 deselected, 33 xfailed, 1 xpassed in 28.64s =====
       For full logs, run:
         nix log /nix/store/1zfm396yb39h8rp3yybhm2h28y6llpvk-python3.14-weasyprint-66.0.drv
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.2, NixOS, 25.11 (Xantusia), 25.11.20251228.9ef2612`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixos-25.05, nixos-unstable"`
 - nixpkgs: `/nix/store/xhgq6c23wj99i1yb77vjb12i95dkbf30-source`

### Notify maintainers

@DutchGerman @friedow 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/primer3/default.nix
+++ b/pkgs/development/python-modules/primer3/default.nix
@@ -8,7 +8,6 @@
   fetchFromGitHub,
   gcc,
   pytestCheckHook,
-  pythonOlder,
   setuptools,
 }:
 
@@ -16,8 +15,6 @@ buildPythonPackage rec {
   pname = "primer3";
   version = "2.2.0";
   pyproject = true;
-
-  disabled = pythonOlder "3.7";
 
   src = fetchFromGitHub {
     owner = "libnano";
@@ -37,6 +34,7 @@ buildPythonPackage rec {
     click
     pytestCheckHook
   ];
+
   # We are not sure why exactly this is need. It seems `pytestCheckHook`
   # doesn't find extension modules installed in $out/${python.sitePackages},
   # and the tests rely upon them. This was initially reported upstream at

--- a/pkgs/development/python-modules/primer3/default.nix
+++ b/pkgs/development/python-modules/primer3/default.nix
@@ -11,7 +11,7 @@
   setuptools,
 }:
 
-buildPythonPackage rec {
+buildPythonPackage (finalAttrs: {
   pname = "primer3";
   version = "2.2.0";
   pyproject = true;
@@ -19,7 +19,7 @@ buildPythonPackage rec {
   src = fetchFromGitHub {
     owner = "libnano";
     repo = "primer3-py";
-    tag = "v${version}";
+    tag = "v${finalAttrs.version}";
     hash = "sha256-GrVYYjS/+LZScZETfk7YcSy2yrWc3SPumXvyQeEpFUg=";
   };
 
@@ -49,8 +49,8 @@ buildPythonPackage rec {
   meta = {
     description = "Oligo analysis and primer design";
     homepage = "https://github.com/libnano/primer3-py";
-    changelog = "https://github.com/libnano/primer3-py/blob/${src.tag}/CHANGES";
+    changelog = "https://github.com/libnano/primer3-py/blob/${finalAttrs.src.tag}/CHANGES";
     license = lib.licenses.gpl2Only;
     maintainers = with lib.maintainers; [ fab ];
   };
-}
+})

--- a/pkgs/development/python-modules/weasyprint/default.nix
+++ b/pkgs/development/python-modules/weasyprint/default.nix
@@ -97,6 +97,8 @@ buildPythonPackage rec {
     "test_visibility_3"
     "test_visibility_4"
     "test_woff_simple"
+    # AssertionError
+    "test_2d_transform"
   ];
 
   FONTCONFIG_FILE = "${fontconfig.out}/etc/fonts/fonts.conf";

--- a/pkgs/development/python-modules/weasyprint/default.nix
+++ b/pkgs/development/python-modules/weasyprint/default.nix
@@ -30,7 +30,7 @@
   writableTmpDirAsHomeHook,
 }:
 
-buildPythonPackage rec {
+buildPythonPackage (finalAttrs: {
   pname = "weasyprint";
   version = "66.0";
   pyproject = true;
@@ -40,7 +40,7 @@ buildPythonPackage rec {
   src = fetchFromGitHub {
     owner = "Kozea";
     repo = "WeasyPrint";
-    tag = "v${version}";
+    tag = "v${finalAttrs.version}";
     hash = "sha256-wmEDVEbikBpOQ5394IBPWQRjWZOLfMzEGxTtq4tt2Tw=";
   };
 
@@ -104,16 +104,16 @@ buildPythonPackage rec {
   FONTCONFIG_FILE = "${fontconfig.out}/etc/fonts/fonts.conf";
 
   # Set env variable explicitly for Darwin, but allow overriding when invoking directly
-  makeWrapperArgs = [ "--set-default FONTCONFIG_FILE ${FONTCONFIG_FILE}" ];
+  makeWrapperArgs = [ "--set-default FONTCONFIG_FILE ${finalAttrs.FONTCONFIG_FILE}" ];
 
   pythonImportsCheck = [ "weasyprint" ];
 
   meta = {
-    changelog = "https://github.com/Kozea/WeasyPrint/releases/tag/${src.tag}";
+    changelog = "https://github.com/Kozea/WeasyPrint/releases/tag/${finalAttrs.src.tag}";
     description = "Converts web documents to PDF";
-    mainProgram = "weasyprint";
     homepage = "https://weasyprint.org/";
     license = lib.licenses.bsd3;
+    mainProgram = "weasyprint";
     teams = [ lib.teams.apm ];
   };
-}
+})

